File: cs/ch17-04-arithmetic/Money/ICurrencyHolder.cs
namespace TheSoftwareGorilla.TDD.Money;
public interface ICurrencyHolder<T> where T : ICurrencyHolder<T>

{
    public Bank<T> Bank { get; }
    public decimal Amount { get; }
    public string Currency { get; }
    public T Convert(string to);
    public decimal AmountIn(string currency);

    public T NewCurrencyHolder(decimal amount, string currency, Bank<T> bank);
}

File: java/ch17-04-arithmetic/src/main/java/com/thesoftwaregorilla/tdd/money/ICurrencyHolder.java
package com.thesoftwaregorilla.tdd.money;

import java.math.BigDecimal;

public interface ICurrencyHolder<T extends ICurrencyHolder<T>> {

    public Bank<T> getBank();
    public String getCurrency();
    public BigDecimal getAmount();
    public T convert(String toCurrency);
    public BigDecimal getAmountIn(String toCurrency);
    public T newCurrencyHolder(BigDecimal amount, String currency, Bank<T> bank);

}
